docker-app/qfieldcloud/core/management/commands/extractstoragemetadata.py

```diff
@@ -0,0 +1,141 @@
+import csv
+import logging
+import sys
+from collections import namedtuple
+from functools import reduce
+from pathlib import Path
+from typing import Any, Generator
+
+import yaml
+from django.core.management.base import BaseCommand, CommandParser
+from qfieldcloud import settings
+from qfieldcloud.core import utils
+
+logger = logging.getLogger(__name__)
+
+s3_credentials_keys = [
+    "STORAGE_ACCESS_KEY_ID",
+    "STORAGE_SECRET_ACCESS_KEY",
+    "STORAGE_BUCKET_NAME",
+    "STORAGE_REGION_NAME",
+    "STORAGE_ENDPOINT_URL",
+]
+
+S3ConfigObject = namedtuple("S3ConfigObject", s3_credentials_keys)
+
+
+class S3Config:
+    _config = None
+
+    @staticmethod
+    def from_file(path_to_config: Path) -> dict[str, Any]:
+        """Load a YAML file exposing credentials used by S3 storage"""
+        with open(path_to_config) as fh:
+            contents = yaml.safe_load(fh)
+        return contents
+
+    @staticmethod
+    def validate(contents: dict[str, Any]) -> dict[str, Any]:
+        """Validate key/values used by S3 storage"""
+        reporter = {"missing_value": [], "extra_key": []}
+
+        def reducer(acc, item):
+            key, val = item
+            if key not in s3_credentials_keys:
+                acc["extra_key"].append(key)
+            if val in ("", None):
+                acc["missing_value"].append(key)
+            return acc
+
+        reported = reduce(reducer, contents.items(), reporter)
+        reported["missing_key"] = [
+            key for key in s3_credentials_keys if key not in contents.keys()
+        ]
+
+        if any(reported.values()):
+            logger.error(f"Invalid config!: {reported}")
+            sys.exit(1)
+
+        return contents
+
+    @classmethod
+    def get_or_load(cls, path_to_file: str | None) -> S3ConfigObject:
+        """Get or create configuation for S3 storage."""
+        if not cls._config:
+            if not path_to_file:
+                logger.error(
+                    "You need to pass a valid path to your YAML configuration file"
+                )
+                sys.exit(1)
+            maybe_valid_config = cls.from_file(path_to_file)
+            valid_config = cls.validate(maybe_valid_config)
+            cls._config = S3ConfigObject(**valid_config)
+
+        return cls._config
+
+    @classmethod
+    def inject_to_settings(cls):
+        """To avoid passing s3 config as parameters"""
+        for key, value in cls._config._asdict().items():
+            setattr(settings, key, value)
```

Comment:
This whole class is unnessesary. You could have just copy-pasted what we already had:

```
def get_s3_bucket(acces_key_id, secret_access_key, region_name, endpoint_url, bucket_name)
    session = boto3.Session(
        aws_access_key_id=acces_key_id,
        aws_secret_access_key=secret_access_key,
        region_name=region_name,
    )
    s3 = session.resource("s3", endpoint_url=endpoint_url)
    bucket = s3.Bucket(bucket_name)
    return bucket
```

Changing the settings like this is dangerous and quite far from what we needed...